File: ios/Theme/IdScan.swift
public class IdScan {
  private let style: Style
  private let target: NSDictionary?
  private let font: Font
  private let button: Button
  private let image: Image
  private let additionalReview: AdditionalReview
  private let idFeedback: IdFeedback
  private let selectionScreen: SelectionScreen
  private let captureScreen: CaptureScreen
  private let reviewScreen: ReviewScreen

  init() {
    self.style = Style()
    
    self.target = self.style.getTarget("idScan")
    self.font = Font()
    self.button = Button(target: self.target)
    self.image = Image(target: self.target)
    self.additionalReview = AdditionalReview(target: self.target)
    self.idFeedback = IdFeedback(target: self.target)
    self.selectionScreen = SelectionScreen(target: self.target)
    self.captureScreen = CaptureScreen(target: self.target)
    self.reviewScreen = ReviewScreen(target: self.target)
  }

  
  public func getShowFaceMatchToIDBrandingImage() -> Bool {
    let key = "isShowFaceMatchToIDBrandingImage"
    if !self.style.exists(self.target, key: key) {
      return false
    }

    return self.target?[key] as! Bool
  }

  public func getHeaderFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "headerFont")
  }

  public func getSubtextFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "subtextFont")
  }

  public func getInactiveTorchImage() -> UIImage? {
    return self.image.getSource("inactiveTorchImage")
  }

  public func getActiveTorchImage() -> UIImage? {
    return self.image.getSource("activeTorchImage")
  }

  public func getFaceMatchToIDBrandingImage() -> UIImage? {
    return self.image.getSource("faceMatchToIDBrandingImage")
  }
  
  public func getButton() -> Button {
    return self.button
  }
  
  public func getAdditionalReview() -> AdditionalReview {
    return self.additionalReview
  }
  
  public func getIdFeedback() -> IdFeedback {
    return self.idFeedback
  }

  public func getSelectionScreen() -> SelectionScreen {
    return self.selectionScreen
  }

  public func getCaptureScreen() -> CaptureScreen {
    return self.captureScreen
  }

  public func getReviewScreen() -> ReviewScreen {
    return self.reviewScreen
  }
}

File: android/src/main/java/com/azifacemobile/theme/IdScan.java
package com.azifacemobile.theme;

import android.graphics.Typeface;

import com.azifacemobile.R;
import com.azifacemobile.utils.Theme;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONException;
import org.json.JSONObject;

import javax.annotation.Nullable;

public class IdScan {
  private final Theme theme;
  private final JSONObject target;
  private final Font font;
  private final Button button;
  private final Image image;
  private final SelectionScreen selectionScreen;
  private final CaptureScreen captureScreen;
  private final ReviewScreen reviewScreen;
  private final AdditionalReview additionalReview;
  private final IdFeedback idFeedback;

  public IdScan(ReactApplicationContext context) {
    this.theme = new Theme();

    this.target = this.theme.getTarget("idScan");
    this.font = new Font(context);
    this.button = new Button(context, this.target);
    this.image =  new Image(context, this.target);
    this.selectionScreen = new SelectionScreen(context, this.target);
    this.captureScreen = new CaptureScreen(context, this.target);
    this.reviewScreen = new ReviewScreen(this.target);
    this.additionalReview = new AdditionalReview(context, this.target);
    this.idFeedback = new IdFeedback(context, this.target);
  }

  public boolean getShowFaceMatchToIDBrandingImage() {
    final String key = "isShowFaceMatchToIDBrandingImage";
    try {
      if (!this.theme.exists(this.target, key)) {
        return false;
      }

      return this.target.getBoolean(key);
    } catch (JSONException e) {
      return false;
    }
  }

  @Nullable
  public Typeface getHeaderFont() {
    return this.font.getTypography(this.target, "headerFont");
  }

  @Nullable
  public Typeface getSubtextFont() {
    return this.font.getTypography(this.target, "subtextFont");
  }

  public Button getButton() {
    return this.button;
  }

  public int getInactiveTorchImage() {
    return this.image.getSource("inactiveTorchImage", R.drawable.facetec_inactive_torch);
  }

  public int getActiveTorchImage() {
    return this.image.getSource("activeTorchImage", R.drawable.facetec_active_torch);
  }

  public int getFaceMatchToIDBrandingImage() {
    return this.image.getSource("faceMatchToIDBrandingImage");
  }

  public SelectionScreen getSelectionScreen() {
    return this.selectionScreen;
  }

  public CaptureScreen getCaptureScreen() {
    return this.captureScreen;
  }

  public ReviewScreen getReviewScreen() {
    return this.reviewScreen;
  }

  public AdditionalReview getAdditionalReview() {
    return this.additionalReview;
  }

  public IdFeedback getIdFeedback() {
    return this.idFeedback;
  }
}
